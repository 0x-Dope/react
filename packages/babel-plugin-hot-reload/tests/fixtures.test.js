'use strict';

const pluginTester = require('babel-plugin-tester');
const hotReload = require('..');
const path = require('path');

pluginTester({
  plugin: hotReload,
  filename: __filename,
  babelOptions: {
    parserOpts: { plugins: ['jsx'] },
    generatorOpts: {},
    babelrc: false,
  },
  snapshot: true,
  tests: [
    {
      title: 'single default',
      fixture: path.join(__dirname, 'fixtures', 'single-default-export.js'),
      snapshot: true,
    },
  ],
});
