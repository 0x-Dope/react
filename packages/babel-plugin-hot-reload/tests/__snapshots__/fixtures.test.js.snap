// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hot-reload assignment: assignment 1`] = `
"
let LazyCC;
LazyCC = 5;

let DoubleLazyCC = LazyCC;

      ↓ ↓ ↓ ↓ ↓ ↓

let LazyCC;
LazyCC = window.__assign(module, \\"LazyCC\\", 5);

let DoubleLazyCC = window.__assign(module, \\"DoubleLazyCC\\", LazyCC);
"
`;

exports[`hot-reload default export class: default export class 1`] = `
"
import React from 'react';
import Hello from './Hello';

export default class App extends React.Component {
  render() {
    return <Hello />;
  }
}

module.hot.accept(['./Hello'], window.__invalidate);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';
import Hello from './Hello';
export default class App extends React.Component {
  render() {
    return <Hello />;
  }

}
module.hot.accept(['./Hello'], window.__invalidate);
"
`;

exports[`hot-reload default export: default export 1`] = `
"
import React from 'react';
import Hello from './Hello';
import Layout from './Layout';
import './App.css';

export default function App({ children }) {
  return (
    <Layout>
      <Hello />
    </Layout>
  );
}

module.hot.accept(['./Layout', './Hello'], window.__invalidate);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';
import Hello from './Hello';
import Layout from './Layout';
import './App.css';

var App = window.__assign(module, \\"App\\", function App({
  children
}) {
  return <Layout>
      <Hello />
    </Layout>;
});

export default App;
module.hot.accept(['./Layout', './Hello'], window.__invalidate);
"
`;

exports[`hot-reload lazy component: lazy component 1`] = `
"
import { lazy } from 'react';
const LazyCC = lazy(() => import('./CounterClass'));

      ↓ ↓ ↓ ↓ ↓ ↓

import { lazy } from 'react';

const LazyCC = window.__assign(module, \\"LazyCC\\", lazy(() => import('./CounterClass')));
"
`;
